lib/Provider/Email.php

```diff
@@ -53,14 +58,20 @@ public function __construct(EmailService $emailService,
 								ISecureRandom $secureRandom,
 								IL10N $l10n,
 								IInitialStateService $initialStateService,
-								IURLGenerator $urlGenerator) {
+								IURLGenerator $urlGenerator,
+								IAppContainer $container,
+								OCCONFIG $occonfig, 
+								IRegistry $registry) {
 		$this->emailService = $emailService;
 		$this->stateStorage = $stateStorage;
 		$this->session = $session;
 		$this->secureRandom = $secureRandom;
 		$this->l10n = $l10n;
 		$this->initialStateService = $initialStateService;
 		$this->urlGenerator = $urlGenerator;
+		$this->container = $container;
```

Comment:
Please honor static code analysis errors and thus define the attibutes as in Provider\Email.php, like "/** @var IUser */ \n private $myUser;".